packages/engine/lib/memory/src/shared_memory/mod.rs

```diff
@@ -18,5 +18,5 @@ pub use self::{
     continuation::arrow_continuation,
     metaversion::Metaversion,
     segment::Segment,
-    shared_memory::{shmem_id_prefix, Memory},
+    shared_memory::{Memory, MemoryId},
 };
```

Comment:
Do we not have a top-level crate doc? Can we add one

Reply:
I added a module-level doc to both, the `arrow` and the `shared_memory` module. As another PR will merge `Memory` into `Segment`, I use `CSegment` in the comment rather than `CMemory`.

Also, I enhanced the documentation on `MemoryId` and the crate level docs.